lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -770,11 +934,11 @@ struct ObjectOpConversion : public OpConversionPattern<om::ObjectOp> {
   LogicalResult
   matchAndRewrite(om::ObjectOp objectOp, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    rewriter.updateRootInPlace(objectOp, [&] {
-      auto operands = adaptor.getOperands();
-      for (unsigned i = 0, e = objectOp->getNumOperands(); i < e; ++i)
-        objectOp->setOperand(i, operands[i]);
-    });
+    // Replace the object with a new object using the converted actual parameter
+    // types from the adaptor.
+    rewriter.replaceOpWithNewOp<om::ObjectOp>(objectOp, objectOp.getType(),
```

Comment:
While resolving merge conflicts, I took the implementation I had over the one on main. I think it does the same thing, but the one-liner is a little simpler. This change isn't technically required, and I am also happy to back it out to keep this patch simpler.